### PR TITLE
[ISSUE #1122] support custom `asyncThreadPool`

### PR DESCRIPTION
**Pull request by dragon-zhang**
see https://github.com/sofastack/sofa-rpc/issues/1122

### CODE REVIEW

**Comment by dragon-zhang** [comment]
111111企业微信截图_2e680565-6575-4875-ab2a-52efaa9e94bc
我明明签了CLA呀

**Comment by seeflood** [comment]
@dragon-zhang cla好啦，感谢贡献。

**Comment by dragon-zhang** [comment]
> @dragon-zhang cla好啦，感谢贡献。

em...master的测试用例有些问题，这个PR只加了设置线程池的方法.

File: core/api/src/main/java/com/alipay/sofa/rpc/context/AsyncRuntime.java [hunk 1]
```diff
@@ -100,4 +100,13 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
...
         return asyncThreadPool;
     }
+
+    /**
+     * 设置callback用的线程池
+     *
+     * @param asyncThreadPool callback用的线程池
+     */
+    public static void setAsyncThreadPool(ThreadPoolExecutor asyncThreadPool) {
+        AsyncRuntime.asyncThreadPool = asyncThreadPool;
```
Comment (JervyShi) [inline core/api/src/main/java/com/alipay/sofa/rpc/context/AsyncRuntime.java:110]: We need to consider a more reasonable configuration for supporting asynchronous thread pool setups. The changes here do not guarantee thread safety and do not ensure that no other logic will have access to the thread pool created by the original logic before the set method is called. I hope to have a modified solution to discuss before putting it into development.

**Review by JervyShi — CHANGES_REQUESTED** [review]

File: core/api/src/main/java/com/alipay/sofa/rpc/context/AsyncRuntime.java (hunk 1 above)
Comment (dragon-zhang) [reply core/api/src/main/java/com/alipay/sofa/rpc/context/AsyncRuntime.java:110]: > We need to consider a more reasonable configuration for supporting asynchronous thread pool setups. The changes here do not guarantee thread safety and do not ensure that no other logic will have access to the thread pool created by the original logic before the set method is called. I hope to have a modified solution to discuss before putting it into development.

sorry, I will follow the steps.

**Pull request closed without merging**